VideoLocker/src/main/java/org/edx/mobile/util/SecurityUtil.java

```diff
@@ -0,0 +1,50 @@
+package org.edx.mobile.util;
+
+
+import android.content.Context;
+import android.content.pm.PackageInfo;
+import android.content.pm.PackageManager;
+import android.support.annotation.NonNull;
+
+import org.edx.mobile.logger.Logger;
+import org.edx.mobile.module.db.DbStructure;
+import org.edx.mobile.module.prefs.PrefManager;
+
+import java.io.File;
+import java.util.Collections;
+
+public abstract class SecurityUtil {
+    private static final Logger logger = new Logger(SecurityUtil.class);
+
+    public static void clearUserData(@NonNull Context context) {
+        // Clear the App's directory
+        PackageManager packageManager = context.getPackageManager();
+        try {
+            PackageInfo packageInfo = packageManager.getPackageInfo(context.getPackageName(), 0);
+            File appDir = new File(packageInfo.applicationInfo.dataDir);
+            if (appDir.exists() && appDir.isDirectory()) {
+                File[] children = appDir.listFiles();
```

Comment:
Note that you can simply get the list of files, and check if the returned array is a `null` value, instead of having to check the existence and directory status of the `File` as well. This would also make this an atomic operation, and not subject to race conditions like your current conditional logic is.